beacon-chain/rpc/core/validator.go

```diff
@@ -264,3 +266,77 @@ func (s *Service) SubmitSignedAggregateSelectionProof(
 
 	return nil
 }
+
+// ProduceSyncCommitteeContribution requests that the beacon node produce a sync committee contribution.
+func (s *Service) ProduceSyncCommitteeContribution(
```

Comment:
This function should probably not be in `core`. This package is meant to contain code that is called both from `eth` and `v1alpha1`, but this one is only needed in `eth`. It can be a private function in the same package as the handler. Using `ProduceSyncCommitteeContributionRequest` as a parameter is wrong because we want to get rid of this type. You can pass its fields directly instead. You also shouldn't return `ProduceSyncCommitteeContributionResponse` for the same reason. When you move this function out of `core`, you won't need to return `RpcError` - what you can do is write the error here and return `(X, bool)` from the function, using `false` as a way to indicate that the function failed. So basically you write something like this:
```
result, ok :=  produceSyncCommitteeContribution()
if !ok {
    return
}
```